integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -327,11 +327,13 @@ private int readInternal(String path, ByteBuffer buf, long size, long offset, Fu
           LOG.error("Cannot find fd {} for {}", fd, path);
           return -ErrorCodes.EBADFD();
         }
-        is.seek(offset);
-        while (rd >= 0 && nread < size) {
-          rd = is.read(dest, nread, sz - nread);
-          if (rd >= 0) {
-            nread += rd;
+        if (offset - is.getPos() < is.remaining()) {
```

Comment:
if offset exceeds the input stream length, according to POSIX definition, we shall still read bytes until file ends.

Reply:
I think if offset is the start point which the current request want to read, but if it exceeds the input stream length, it means no bytes to read, correct me if I'm wrong?